hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/HTSPrecompiledContract.java

```diff
@@ -636,12 +667,12 @@ HederaTokenStore newHederaTokenStore(
 					validator,
 					null,
 					recordsHistorian);
+
 			for (int i = 0, n = changes.size(); i < n; i++) {
 				final var change = changes.get(i);
 				final var units = change.units();
 				if (change.isForNft() || units < 0) {
-					final var hasSenderSig = sigsVerifier.hasActiveKey(
-							change.getAccount(), recipient, contract);
+					final var hasSenderSig = validateKey(frame, change.getAccount().asEvmAddress(), sigsVerifier::hasActiveKey);
```

Comment:
Is all this Key validation logic leaking in from another PR or is it somehow relevant to the gas cost calculations?

Reply:
I should read slack before reviewing PRs.  This has changes from #2728.

Reply:
Yes, these changes are part of #2728, I decided to merge them here to ease up resolving of conflicts later, since both change similar classes... maybe I should have kept them separate for ease of readability